src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/BuildHostProcessManager.cs

```diff
@@ -33,22 +33,41 @@ public async Task<IBuildHost> GetBuildHostAsync(string projectFilePath, Cancella
     {
         var neededBuildHostKind = GetKindForProject(projectFilePath);
 
-        _logger?.LogTrace($"Choosing a build host of type {neededBuildHostKind} for {projectFilePath}");
+        _logger?.LogTrace($"Choosing a build host of type {neededBuildHostKind} for {projectFilePath}.");
 
+        var buildHost = await GetBuildHostAsync(neededBuildHostKind, cancellationToken).ConfigureAwait(false);
+
+        // If this is a .NET Framework build host, we may not have have build tools installed and thus can't actually use it to build.
+        // Check if this is the case.
+        if (neededBuildHostKind == BuildHostProcessKind.NetFramework)
+        {
+            if (!await buildHost.IsProjectFileSupportedAsync(projectFilePath, cancellationToken))
+            {
+                // It's not usable, so we'll fall back to the .NET Core one.
+                _logger?.LogWarning($"An installation of Visual Studio or the Build Tools for Visual Studio could not be found; {projectFilePath} will be loaded with the .NET Core SDK and may encounter errors.");
```

Comment:
we should consider adding a doc on how to acquire these tools - maybe after we finish adding mono support as well.

Reply:
We might have to write the Mono docs sooner, just because we'll need it for testing!